src/routes/auth.js

```diff
@@ -388,3 +391,77 @@ router.post(`/sudoTestCommand/:secret/makeadminforclient`, async (req, res) => {
         });
     }
 });
+
+router.post('/requestToken', async (req, res) => {
+    try {
+        const { jwt } = req.body;
+        const { clientId } = decode(jwt);
+        const client = await Client.findById(clientId);
+
+        if (!client) {
+            return res.status(400).json({
+                err: true,
+                msg: 'No client found',
+            });
+        }
+
+        verify(jwt, client.access_token, {
+            algorithms: ['HS256'],
+        });
+        rtoken.exists = util.promisify(rtoken.exists);
+        let requestToken;
+        let exists = 1;
+        while (exists) {
+            requestToken = makeid(64, true);
+            // eslint-disable-next-line no-await-in-loop
+            exists = await rtoken.exists(requestToken.toString());
+        }
+        rtoken.hmset(requestToken.toString(), { cId: clientId });
+        rtoken.expire(requestToken.toString(), keys.reqExpTime);
+        const token = getRequestToken(requestToken);
+        res.send(token);
+    } catch (error) {
+        console.log(error);
+        return res.status(401).json({
+            err: true,
+            msg: 'Unauthorized Client',
+        });
+    }
+});
+
+router.get('/verifyRToken', async (req, res) => {
+    try {
+        console.log('here');
+        const { q } = req.query;
+        const { requestToken } = decode(q);
+        rtoken.exists = util.promisify(rtoken.exists);
+        rtoken.hget = util.promisify(rtoken.hget);
+        const exists = await rtoken.exists(requestToken.toString());
+        console.log(exists);
+        if (!exists) {
+            return res.status(401).json({
+                err: true,
+                msg: 'Session Expired',
+            });
+        }
+        const user = await verifyToken(req, res);
+        const clientId = await rtoken.hget(requestToken.toString(), 'cId');
```

Comment:
You will need to add 2 more checks before storing the user auth token in redis.
1. Check if JWT token in the query parameter `q` is signed by the corresponding client access token.
2. Check if the client ID stored in redis corresponding to the request token matches the client id in the JWT payload.

Reply:
Second check cannot be implemented since the JWT payload only contains the requestToken and not the clientID.